Based on Peter Korsgaard <peter@korsgaard.com> patch from nezha
---
 Makefile  | 9 +++++++++
 board.cfg | 9 +++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 board.cfg

diff --git a/Makefile b/Makefile
index f911f70344..259d93bf80 100644
--- a/Makefile
+++ b/Makefile
@@ -1084,6 +1084,15 @@ endif
 .binman_stamp: FORCE
 	@touch $@
 
+fw_dynamic.bin: $(OPENSBI)
+	$(call if_changed,copy)
+
+MKIMAGEFLAGS_u-boot.toc1 = -T sunxi_toc1
+u-boot.toc1: board.cfg fw_dynamic.bin inputs
+	$(call if_changed,mkimage)
+
+all: u-boot.toc1
+
 ifeq ($(CONFIG_DEPRECATED),y)
 	$(warning "You have deprecated configuration options enabled in your .config! Please check your configuration.")
 endif
diff --git a/board.cfg b/board.cfg
new file mode 100644
index 0000000000..2d23b9b388
--- /dev/null
+++ b/board.cfg
@@ -0,0 +1,9 @@
+[opensbi]
+file = fw_dynamic.bin
+addr = 0x40000000
+[dtb]
+file = arch/riscv/dts/board.dtb
+addr = 0x44000000
+[u-boot]
+file = u-boot-nodtb.bin
+addr = 0x4a000000
-- 
2.20.1

